modules/vpc-sc/variables.tf

```diff
@@ -66,94 +67,103 @@ variable "access_policy_create" {
   default = null
 }
 
+variable "egress_policies" {
+  description = "Egress policy definitions that can be referenced in perimeters."
+  type = map(object({
+    from = object({
+      identity_type = optional(string, "ANY_IDENTITY")
+      identities    = optional(list(string))
+    })
+    to = object({
+      operations = optional(list(object({
+        method_selectors = optional(list(string))
+        service_name     = string
+      })), [])
+      resources              = optional(list(string))
+      resource_type_external = optional(bool, false)
+    })
+  }))
+  default  = {}
+  nullable = false
+  validation {
+    condition = alltrue([
+      for k, v in var.egress_policies : contains([
+        "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
+        "ANY_USER", "ANY_SERVICE_ACCOUNT"
+      ], v.from.identity_type)
+    ])
+    error_message = "Invalid `from.identity_type` value in eress policy."
+  }
+}
+
+variable "ingress_policies" {
+  description = "Ingress policy definitions that can be referenced in perimeters."
+  type = map(object({
+    from = object({
+      access_levels = optional(list(string), [])
+      identity_type = optional(string)
+      identities    = optional(list(string))
+      resources     = optional(list(string), [])
+    })
+    to = object({
+      operations = optional(list(object({
+        method_selectors = optional(list(string))
+        service_name     = string
+      })), [])
+      resources = optional(list(string))
+    })
+  }))
+  default  = {}
+  nullable = false
+  validation {
+    condition = alltrue([
+      for k, v in var.ingress_policies :
+      v.from.identity_type == null || contains([
+        "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
+        "ANY_USER", "ANY_SERVICE_ACCOUNT"
+      ], coalesce(v.from.identity_type, "-"))
```

Comment:
I think the coalesce here is not needed (the branch of the OR guarantees it always non-null)

Reply:
hah, I got an error on null type and had to ad the coalesce :)